src/main/java/org/tools4j/hoverraft/transport/aeron/AeronReceiver.java

```diff
@@ -27,14 +27,15 @@
 import io.aeron.logbuffer.FragmentHandler;
 import io.aeron.logbuffer.Header;
 import org.agrona.DirectBuffer;
+import org.tools4j.hoverraft.message.direct.AbstractDirectMessage;
 import org.tools4j.hoverraft.message.direct.DirectMessage;
 import org.tools4j.hoverraft.message.direct.DirectMessageFactory;
 import org.tools4j.hoverraft.transport.Receiver;
 
 import java.util.Objects;
 import java.util.function.Consumer;
 
-public class AeronReceiver implements Receiver<DirectMessage> {
+public class AeronReceiver implements Receiver<AbstractDirectMessage> {
```

Comment:
no way --- i really don't like that AbstractDirectMessage is now exposed everywhere.